### ISSUE TITLE
Allow other browsers

### ISSUE THREAD

**Issue by piranna** [GoogleChromeLabs/carlo#27]
Allow to use other browsers if Chrome is not available. Maybe all features are not available, but for just launching a web page that connect to de Node.js server would be far from enough... :-)

**Comment by KeziahMoselle** [comment]
It use Puppeteer so I don't think they can support others browsers than Chrome :(

**Comment by nklayman** [comment]
The other browsers would have to implement puppeteer for this to work. Until then, I think this will be labeled as `wontfix`.

**Comment by fenwick67** [comment]
Other browsers support WebDriver instead of puppeteer for browser control, so I believe it's theoretically possible with some extra glue.

**Comment by Xotabu4** [comment]
@fenwick67 WebDriver is much slower because uses http REST, and Chrome Debug Protocol uses websockets. There will be higher latency. And WebDriver protocol does not have all the features of Chrome Debug Protocol (for example Network, Performance and other - https://chromedevtools.github.io/devtools-protocol/)

**Comment by fenwick67** [comment]
@Xotabu4 Yes, webdriver is transactional HTTP, which will be slower than websockets for things like opening or moving a window (though I'm skeptical it would be perceptible in most cases, since it's all over loopback anyway). 

But I think it should be possible to load some js into the window to handle websocket connections for more performance-critical RPC stuff, as long as the functionality could be done with the regular browser APIs.

**Comment by ab35** [comment]
OS manufacturers should work on a browser shell standard common to all operating systems, because after all, all browsers do the same thing according to W3C standards.

So particular brands of browsers like Edge, Safari, Chrome, etc would all derive from a common standard shell present in all operating systems.

Then, users would just have to choose to what browser brand they want to upgrade / or there will be one brand by default installed along the shell in the particular OS (for example the Edge brand in windows OS).

This would allow to greatly reduce the internet bandwidth when acquiring and updating new browsers on an OS, and carlo apps wouldn't need to ship with chromium while still being compatible with all browser brands.

**Comment by KeziahMoselle** [comment]
This is a solution for Firefox :
[https://github.com/GoogleChrome/puppeteer/tree/master/experimental/puppeteer-firefox#readme](https://github.com/GoogleChrome/puppeteer/tree/master/experimental/puppeteer-firefox#readme)

**Comment by sbhamad** [comment]
Behold the next generation of puppeteer. Works with all major browsers, and has more features. @piranna 
 
https://github.com/microsoft/playwright
